pkg/http/round_tripper_with_headers_test.go

```diff
@@ -19,7 +19,7 @@ func TestRoundTripperWithHeaders(t *testing.T) {
 		expectedHeaderFields: expectedHeaderFields,
 	}
 
-	rtWithHeaders := slinkyhttp.NewRoundTripperWithHeaders(expectedHeaderFields, rt)
+	rtWithHeaders := slinkyhttp.NewRoundTripperWithHeaders(rt, slinkyhttp.WithAuthentication("X-Api-Key", "test"))
```

Comment:
Can we just add the expected user agent header here? Or is it difficult to test because it includes the commit sha on non-releases?